HotelBooking/HotelBooking.App/Operations/OperationBase.cs

```diff
@@ -0,0 +1,36 @@
+﻿using System;
+using HotelBooking.Domain.Models;
+using HotelBooking.App.DataObjects;
+using HotelBooking.App.Enums;
+
+namespace HotelBooking.App.Operations
+{
+    public abstract class OperationBase : IOperation
+    {
+        public abstract bool IsRequiredToSucceed { get; }
+        public abstract string OperationName { get; }
+        public abstract Func<Reservation, bool> Execute { get; }
+
+        public OperationResult OperationResult { get; set; }
+
+        public void Process(Reservation reservation, BookingResult bookingResult)
+        {
+            var executionResult = Execute(reservation) ? Result.Success : Result.Failure;
+
+            OperationResult = new OperationResult
+            {
+                ExecutionResult = executionResult,
+                OperationName = OperationName,
+                ShouldAbortBookingProcess = IsRequiredToSucceed && executionResult == Result.Failure
+            };           
+            bookingResult.OperationResults.Add(OperationResult);
+
+            WriteOperationResultToConsole();            
+        }
+
+        private void WriteOperationResultToConsole()

```

Comment:
To cos co napewno warto zrobic. W wielu miejsach wstawiasz sobie tak lekka reka odwolanie do konsoli, tym samym uzalezniajac corowe klasy twojej apki od takiej infrastrukturalnej rzeczy jak consola. Co mozna zrobic.

W tym konkretnym przypadku zwrocilbym rezultat moze nawet do pliku programu i tam mialbym jakiegos formattera ktory by mi to sformatowal na cosole. W innych przypdkach, tam gdzie wykorzystuje ta informacje jak DEBUG INFO wstrzyknolbym jako zalerznosc cos w stylu ILogger albo IDebugInfoBuilder